lib/flare_animation.dart

```diff
@@ -0,0 +1,133 @@
+import 'dart:ui';
+import 'dart:math';
+
+import "flame.dart";
+import "package:flare_flutter/flare.dart";
+import "package:flare_flutter/flare_actor.dart";
+
+class FlareAnimation {
+  final FlutterActorArtboard _artboard;
+
+  String _animationName;
+  final List<FlareAnimationLayer> _animationLayers = [];
+
+  double x = 0.0,
+      y = 0.0,
+      _width = 0.0,
+      _height = 0.0,
+      xScale = 0.0,
+      yScale = 0.0;
+
+  Picture _picture;
+
+  FlareAnimation(this._artboard);
+
+  static load(String fileName) async {
+    final actor = FlutterActor();
+    await actor.loadFromBundle(Flame.bundle, fileName);
+
+    final artboard = actor.artboard.makeInstance();
+    artboard.makeInstance();
+    artboard.initializeGraphics();
+    artboard.advance(0.0);
+
+    return FlareAnimation(artboard);
+  }
+
+  double get width {
+    return _width;
+  }
+
+  double get height {
+    return _height;
+  }
+
+  set width(double newWidth) {
+    _width = newWidth;
+    xScale = _width / _artboard.width;
+  }
+
+  set height(double newHeight) {
+    _height = newHeight;
+    yScale = _height / _artboard.height;
+  }
+
+  void updateAnimation(String animation) {
+    _animationName = animation;
+
+    if (_animationName != null && _artboard != null) {
+      _animationLayers.clear();
+
+      ActorAnimation animation = _artboard.getAnimation(_animationName);
+      if (animation != null) {
+        _animationLayers.add(FlareAnimationLayer()
+          ..name = _animationName
+          ..animation = animation
+          ..mix = 1.0
+          ..mixSeconds = 0.2);
+        animation.apply(0.0, _artboard, 1.0);
+        _artboard.advance(0.0);
+      }
+    }
+  }
+
+  @override
+  void render(Canvas canvas) {
+    canvas.save();
```

Comment:
you don't need to do that on each component, basegame will take care of saving and restoring the canvas

Reply:
Yeah, but I don't think it is cool to leave "side-effects" like that on the canvas, this may lead to confusing behaviour.

Reply:
But it would be a huge hassle to have to add that to every single component, that's why the (single) game class does it instead, which I think is much cleaner. But since this is no longer a component, you can decide which you like the best, for example, sprite & animation classes don't interfere with the canvas (they use methods that don't add to the matrix). So that's a precedent for keeping here. I'd advise against add that code explicitly to the component class.